dolphin/api/v1/storages.py

```diff
@@ -42,11 +47,64 @@ def validate_parameters(data, required_parameters,
             msg = _("Required parameter %s is empty.") % parameter
             raise exc_response(explanation=msg)
 
+LOG = log.getLogger(__name__)
+
+
+def build_storages(storages):
+    # Build list of storages
+    views = [build_storage(storage)
+             for storage in storages]
+    return dict(storages=views)
+
+
+def build_storage(storage):
+    view = copy.deepcopy(storage)
+    return view
+
 
 class StorageController(wsgi.Controller):
 
     def index(self, req):
-        return dict(name="Storage 1")
+
+        supported_filters = [
+            'name',
+            'vendor',
+            'model',
+            'status',
+        ]
+        filters = {}
+        filters.update(req.GET)
+        # update options  other than filters
+        if 'sort_key' in filters:
+            sort_key = [filters['sort_key']]
+        else:
+            sort_key = None
+        if 'sort_dir' in filters:
+            sort_dir = [filters['sort_dir']]
+        else:
+            sort_dir = None
+        if 'limit' in filters:
+            limit = filters['limit']
+        else:
+            limit = None
+        if 'offset' in filters:
+            offset = filters['offset']
+        else:
+            offset = None
+        # strip out options except supported filter options
+        unknown_options = [opt for opt in filters
+                           if opt not in supported_filters]
+        bad_options = ", ".join(unknown_options)
+        LOG.debug("Removing options '%(bad_options)s' from query",
+                  {"bad_options": bad_options})
+        for opt in unknown_options:
+            del filters[opt]
+        try:
+            storages = db.storage_get_all(context, None, limit, sort_key, sort_dir, filters, offset)
```

Comment:
In DB APIs, both sort_keys and sort_dirs are list, so this calling way may need some modification.

Reply:
both are considered as list, renamed accrodigly..